// primevue/stepper/style/stepperstyle.esm.js
import BaseStyle from 'primevue/base/style';

// .p-stepper {
//     overflow-x: auto;
// }
var css = "\n@layer primevue {\n    .p-stepper {\n        position: relative;\n\n    }\n\n    .p-stepper.p-stepper-vertical {\n        display: flex;\n    }\n\n    .p-stepper-nav-content {\n        position: relative;\n    }\n\n    .p-stepper.p-stepper-vertical  .p-stepper-nav-content {\n        display: flex;\n    }\n\n    .p-stepper .p-stepper-list {\n        display: flex;\n        margin: 0;\n        padding: 0;\n        list-style-type: none;\n    }\n\n    .p-stepper-vertical .p-stepper-list {\n        flex-direction: column;\n    }\n\n    .p-stepper-header {\n        position: relative;\n        display: flex;\n        flex: 1 1 auto;\n        overflow: hidden;\n    }\n\n    .p-stepper-horizontal .p-stepper-header {\n        align-items: center;\n        justify-content: center;\n    }\n\n    .p-stepper-vertical .p-stepper-header {\n        flex-direction: column;\n        align-self: flex-start;\n    }\n\n    .p-stepper-header .p-stepper-action {\n        display: inline-flex;\n        flex-direction: column;\n        align-items: center;\n        overflow: hidden;\n        text-decoration: none;\n        cursor: pointer;\n    }\n\n    .p-stepper.p-stepper-readonly .p-stepper-header {\n        cursor: auto;\n    }\n\n    .p-stepper-header.p-highlight .p-stepper-action {\n        cursor: default;\n    }\n\n    .p-stepper-title {\n        display: block;\n        white-space: nowrap;\n        overflow: hidden;\n        text-overflow: ellipsis;\n        max-width: 100%;\n    }\n\n    .p-stepper-number {\n        display: flex;\n        align-items: center;\n        justify-content: center;\n    }\n}\n";
var classes = {
  root: function root(_ref) {
    var props = _ref.props;
    return ['p-stepper p-component', {
      'p-stepper-horizontal': props.orientation === 'horizontal',
      'p-stepper-vertical': props.orientation === 'vertical'
    }];
  },
  navcontent: 'p-stepper-nav-content',
  nav: 'p-stepper-list',
  step: {
    header: function header(_ref2) {
      var instance = _ref2.instance,
        step = _ref2.step,
        index = _ref2.index;
      return ['p-stepper-header', instance.getStepProp(step, 'headerClass'), {
        'p-highlight': instance.isStepActive(index),
        'p-disabled': instance.getStepProp(step, 'disabled')
      }];
    },
    action: 'p-stepper-action',
    step: 'p-stepper-number',
    title: 'p-stepper-title',
    content: function content(_ref3) {
      var instance = _ref3.instance,
        step = _ref3.step;
      return ['p-stepper-step', instance.getStepProp(step, 'contentClass')];
    }
  },
  stepContainer: 'p-stepper-steps'
};
var StepperStyle = BaseStyle.extend({
  name: 'stepper',
  css: css,
  classes: classes
});

export { StepperStyle as default };
